cli/src/commands/branch.rs

```diff
@@ -393,6 +431,80 @@ fn cmd_branch_set(
     Ok(())
 }
 
+fn cmd_branch_move(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchMoveArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let repo = workspace_command.repo().as_ref();
+    let view = repo.view();
+
+    let branch_names = {
+        let is_source_commit = if !args.from.is_empty() {
+            workspace_command
+                .parse_union_revsets(&args.from)?
+                .evaluate()?
+                .containing_fn()
+        } else {
+            Box::new(|_: &CommitId| true)
+        };
+        if !args.names.is_empty() {
+            find_branches_with(&args.names, |pattern| {
+                view.local_branches_matching(pattern)
+                    .filter(|(_, target)| target.added_ids().any(&is_source_commit))
+                    .map(|(name, _)| name.to_owned())
+            })?
+        } else {
+            view.local_branches()
+                .filter(|(_, target)| target.added_ids().any(&is_source_commit))
+                .map(|(name, _)| name.to_owned())
+                .collect()
+        }
+    };
+    let target_commit = workspace_command.resolve_single_rev(&args.to)?;
+
+    if branch_names.is_empty() {
+        writeln!(ui.status(), "No branches to update.")?;
+        return Ok(());
+    }
+    if branch_names.len() > 1 {
+        writeln!(
+            ui.warning_default(),
+            "Updating multiple branches: {}",
```

Comment:
(May be out of scope for this PR)

I wonder whether we should hint that `jj branch set` should be used to move just one branch in cases where `jj branch move --from mybranch` moves multiples.

I'm worried that people might not initially understand the difference with `set`, will only learn this command, and then be very confused when it does the wrong thing.

Perhaps this hint should be conditional of the revision being a branch name.

Reply:
Makes sense. I added a hint to "specify branch by name". 

(As I said, my plan is to replace the current typo-safe `branch set` use case with `branch move`.)

Reply:
Ah, I'm sorry, I misunderstood what `move` does. I missed the "if branch names are specified...".

This makes a lot more sense now, I'll rethink things with this in mind.